include/alpaka/mem/view/ViewConst.hpp

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2022 Benjamin Worpitz, Matthias Werner, Bernhard Manfred Gruber
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/dev/Traits.hpp>
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/idx/Traits.hpp>
+#include <alpaka/mem/view/Traits.hpp>
+#include <alpaka/mem/view/ViewAccessOps.hpp>
+#include <alpaka/mem/view/ViewPlainPtr.hpp>
+#include <alpaka/offset/Traits.hpp>
+#include <alpaka/vec/Vec.hpp>
+
+namespace alpaka
+{
+    //! A non-modifiable wrapper around a view
+    template<typename TView>
+    struct ViewConst : public internal::ViewAccessOps<ViewConst<TView>>
+    {
+        ALPAKA_FN_HOST_ACC ViewConst(TView&& view) : m_view(std::forward<TView>(view))
+        {
+        }
+
+        TView m_view;
+    };
+
+    namespace trait
+    {
+        template<typename TView>
+        struct DevType<ViewConst<TView>> : DevType<TView>
+        {
+        };
+
+        template<typename TView>
+        struct GetDev<ViewConst<TView>> : GetDev<TView>
+        {
+        };
+
+        template<typename TView>
+        struct DimType<ViewConst<TView>> : DimType<TView>
+        {
+        };
+
+        template<typename TView>
+        struct ElemType<ViewConst<TView>>
+        {
+            // const qualify the element type of the inner view
+            using type = typename ElemType<TView>::type const;
+        };
```

Comment:
Here we just `const` qualify the element type. This should make `internal::ViewAccessOps<ViewConst<TView>>` work.